### PR TITLE
Darreldonald/commithistory

### PR DESCRIPTION
**Pull request by DarrelDonald**
Commit history tab hidden for graduation
new commit history button in devolop tab command bar
     opens commit history for detector in devops in a new tab

### CODE REVIEW

**Review by XiaoxuHU — APPROVED** [review]

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 1]
```diff
@@ -616,6 +620,10 @@ export class OnboardingFlowComponent implements OnInit {
     this.gistDialogHidden = true;
   }
 
+  openCommitHistory(){
+    window.open(this.commitHistoryLink);
```
Comment (ShekharGupta1988) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts:624]: does this open in a new tab?

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html [hunk 2]
```diff
@@ -260,7 +260,8 @@ <h5>{{currentTime}}</h5>
         (click)="getDevOptions()"></fab-command-bar-item>
       <fab-command-bar-item key="gistVersion" text="Gist version" [iconProps]="{ iconName: 'BackToWindow' }"
         (click)="showGistDialog()"></fab-command-bar-item>
-      
+      <fab-command-bar-item [style]="commitHistoryVisibilityStyle" key="commitHistory" text="commitHistory" [iconProps]="{iconName: 'ReadingMode'}"
```
Comment (ShekharGupta1988) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html:263]: nitpick. Text should be 'Commit History'

**Review by ShekharGupta1988 — APPROVED** [review]

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 3]
```diff
@@ -409,6 +411,8 @@ export class OnboardingFlowComponent implements OnInit {
     this.diagnosticApiService.getDevopsConfig(`${this.resourceService.ArmResource.provider}/${this.resourceService.ArmResource.resourceTypeName}`).subscribe(devopsConfig => {
       this.detectorGraduation = devopsConfig.graduationEnabled;
       this.deleteVisibilityStyle = !(this.detectorGraduation === true && this.mode !== DevelopMode.Create) ? {display: "none"} : {};
+      this.commitHistoryVisibilityStyle = !(this.detectorGraduation === true && this.mode !== DevelopMode.Create) ? {display: "none"} : {};
+      this.commitHistoryLink = (this.detectorGraduation && this.mode !== DevelopMode.Create) ? `https://dev.azure.com/${devopsConfig.organization}/${devopsConfig.project}/_git/${devopsConfig.repository}?path=/${this.id}/${this.id}.csx&_a=history` : "";
```
Comment (rekhaswaminathan) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts:415]: Make sure to test it out with teams that have folder path in their repo. In those cases, the path in the URL may be different and you need to construct it accordingly. I left the same comment in the backend PR

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts (hunk 1 above)
Comment (DarrelDonald) [reply AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts:624]: yes

**Review by rekhaswaminathan — APPROVED** [review]

**Comment by rekhaswaminathan** [comment]
@DarrelDonald  - Just FYI: Since this PR is dependent on backend API, please merge this once the backend payload is deployed. Otherwise the UI will be broken.

**Pull request merged**